galley/pkg/server/components/validation.go

```diff
@@ -15,36 +15,167 @@
 package components
 
 import (
-	"k8s.io/client-go/kubernetes"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
 
-	"istio.io/istio/galley/pkg/crd/validation"
 	"istio.io/istio/galley/pkg/server/process"
+	"istio.io/istio/mixer/pkg/validate"
 	"istio.io/istio/pkg/cmd"
+	"istio.io/istio/pkg/config/schemas"
+	"istio.io/istio/pkg/webhooks/validation/controller"
+	"istio.io/istio/pkg/webhooks/validation/server"
+
+	"istio.io/pkg/log"
 	"istio.io/pkg/probe"
 )
 
-// NewValidation returns a new validation component.
-func NewValidation(kubeInterface kubernetes.Interface, kubeConfig string,
-	params *validation.WebhookParameters, liveness, readiness probe.Controller) process.Component {
+// This is for lint fix
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func monitorReadiness(stop <-chan struct{}, port uint, readiness *probe.Probe) {
+	const httpsHandlerReadinessFreq = time.Second
+
+	ready := false
+	client := &http.Client{
+		Timeout: time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	for {
+		if err := webhookHTTPSHandlerReady(client, port); err != nil {
```

Comment:
I realize this is probably copied over, but why do we not just check readiness when k8s calls the readiness probe?

Reply:
As part of the readiness http handler? Galley is currently using a file-based readiness probe which periodically checks a file modification time. We verify the https server is running to update (or not) that file.

We'll need to unify readiness and liveness across Pilot and Galley when running as a single service. That might be a good time to hook this up directly to the k8s readiness check.